.github/workflows/golangci-test.yml

```
@@ -0,0 +1,32 @@
+name: golangci-test
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
```

Can this run with only read permissions?
```suggestion

permissions:
    contents: read
```

---

Yes. 
Included permissions to only read and tested.